src/equations/numerical_fluxes.jl

```diff
@@ -0,0 +1,90 @@
+
+# This file contains general numerical fluxes that are not specific to certain equations
+
+"""
+    flux_central(u_ll, u_rr, orientation, equations::AbstractEquations)
+
+The classical central numerical flux `f((u_ll) + f(u_rr)) / 2`. When this flux is
+used as volume flux, the discretization is equivalent to the classical weak form
+DG method (except floating point errors).
+"""
+@inline function flux_central(u_ll, u_rr, orientation, equations::AbstractEquations)
+  # Calculate regular 1D fluxes
+  f_ll = flux(u_ll, orientation, equations)
+  f_rr = flux(u_rr, orientation, equations)
+
+  # Average regular fluxes
+  return 0.5 * (f_ll + f_rr)
+end
+
+
+"""
+    FluxPlusDissipation(numerical_flux, dissipation)
+
+Combine a `numerical_flux` with a `dissipation` operator to create a new numerical flux.
+"""
+struct FluxPlusDissipation{NumericalFlux, Dissipation}
+  numerical_flux::NumericalFlux
+  dissipation::Dissipation
+end
+
+@inline function (numflux::FluxPlusDissipation{NumericalFlux, Dissipation})(u_ll, u_rr, orientation, equations) where {NumericalFlux, Dissipation}
```

Comment:
Why the `where ...` part? Is it because of [this](https://docs.julialang.org/en/v1/manual/performance-tips/#Be-aware-of-when-Julia-avoids-specializing)?

Reply:
Same question applies to the other callable structs below

Reply:
That's what I had in mind but benchmarking shows that it's not necessary here. I've removed the `where` clause in 281f77c6.

Reply:
:+1: